### PR TITLE
Add the global var location into BIR

### LINKED ISSUE #35479: Go to definition doesn't work for public variables in another module
**Description:**
$title.

**Steps to reproduce:**
```ballerina
import ballerina/websubhub;

public function main() {
    error? x = websubhub:INTERNAL_SUBSCRIPTION_ERROR;

    var y = websubhub:UNSUBSCRIPTION_ACCEPTED;
}
```

Both of these are public, final variables.

**Affected Versions:**
2201.0.1

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose

Fixes #35479

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java [hunk 1]
```diff
@@ -152,6 +152,7 @@ private void writeReferencedTypes(ByteBuf buf, List<BType> referencedTypes) {
     private void writeGlobalVars(ByteBuf buf, BIRTypeWriter typeWriter, List<BIRGlobalVariableDcl> birGlobalVars) {
         buf.writeInt(birGlobalVars.size());
         for (BIRGlobalVariableDcl birGlobalVar : birGlobalVars) {
+            writePosition(buf, birGlobalVar.pos);
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java:155]: We'll have to bump the bir version

Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java:155]: Change

**Review by pcnfernando — DISMISSED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java [hunk 2]
```diff
@@ -825,6 +825,7 @@ private void defineServiceDeclarations(DataInputStream inputStream) throws IOExc
     }
 
     private void definePackageLevelVariables(DataInputStream dataInStream) throws IOException {
+        Location pos = readPosition(dataInStream);
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java:828]: We can probably add a BIR test for this too. e.g., https://github.com/ballerina-platform/ballerina-lang/pull/36457/files#diff-159125135700a3223283f43c74293e9213721379205be2e3fe12fe2af28b3883R477 onward.

Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java:833]: Added a test

File: tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/globalvar/test_public_variable_pos.bal [hunk 3]
```diff
@@ -0,0 +1 @@
+import testorg/publicVariables as _;
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/globalvar/test_public_variable_pos.bal:1]: Let's add the licence header.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bala/globalvar/PublicVariableBalaTest.java [hunk 4]
```diff
@@ -68,4 +78,20 @@ public void testModulePublicVariableAccessNegative() {
         validateError(compileResultNegetive, index++, "undefined symbol 'number'", 20, 20);
         assertEquals(compileResultNegetive.getErrorCount(), index);
     }
+
+    @Test(description = "Test the position of global variable available in a module.")
+    public void testModulePublicVariablePos() {
+        CompileResult birTestResult = BCompileUtil.compile(
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bala/globalvar/PublicVariableBalaTest.java:84]: This is just a normal compilation result, so `birTestResult` is kinda misleading?

**Review by MaryamZi — COMMENTED** [review]
Let's update the PR description to mention the fix (i.e., adding the location to the BIR).

**Comment by KavinduZoysa** [comment]
> 

Added a test

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bala/globalvar/PublicVariableBalaTest.java (hunk 4 above)
Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bala/globalvar/PublicVariableBalaTest.java]: Changed

**Review by IMS94 — DISMISSED** [review]
LGTM!

**Comment by MaryamZi** [comment]
@KavinduZoysa, there seem to be test failures.

**Review by IMS94 — APPROVED** [review]
LGTM!

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**